packages/seacas/libraries/ioss/src/Ioss_DynamicTopology.C

```diff
@@ -36,14 +36,463 @@
+
+  bool internalDecompSpecified = internal_decomp_specified(properties);
+  std::string message;
+  int exists = ::file_exists(util, filename, message, internalDecompSpecified);
+
+  int par_size = util.parallel_size();
+  int par_rank = util.parallel_rank();
+
+  if( exists > 0 && exists < par_size ) {
+    std::ostringstream errmsg;
+    errmsg << "ERROR: Unable to open input database";
+    if(par_rank == 0) {
+      errmsg << " '" << filename << "'" << "\n\ton processor(s): " << message;
+    } else {
+      errmsg << ". See processor 0 output for more details.\n";
+    }
+    IOSS_ERROR(errmsg);
+  }
+
+  if( exists == par_size ) {
+    db = Ioss::IOFactory::create(db_type, filename, Ioss::QUERY_TIMESTEPS_ONLY,
+                                 util.communicator(), properties);
+    int bad_count = 0;
+    std::string error_message;
+    bool is_exodus_file = db != nullptr && db->ok(false, &error_message, &bad_count);
+    if(is_exodus_file) {
+      return std::make_pair(filename, db);
+    } else {
+      delete db;
+      std::ostringstream errmsg;
+      errmsg << error_message;
+      errmsg << __FILE__ << ", " << __FUNCTION__ << ", filename " << filename << " is not an exodus file\n";
+      IOSS_ERROR(errmsg);
+    }
+  }
+
+  // Does not exist on any processors
+  return std::make_pair(std::string(), db);
+}
+
+void locate_cyclic_multi_db_state(Ioss::DatabaseIO* db, double targetTime,
+                                  const std::string& ioDB, const std::string& dbType,
+                                  DatabaseState& loc)
+{
+  auto util = db->util();
+  FileNameGenerator generator = [](const std::string& baseFileName, unsigned step) {
+                                     static std::string suffix = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+                                     std::string filename = baseFileName;
+                                     if(step == 0) step++;
+                                     filename += "-" + suffix.substr((step-1)%26, 1);
+                                     return filename;
+                                  };
+  bool found = true;
+  int step = 0;
+  while(found) {
+    ++step;
+
+    std::string expanded;
+    Ioss::DatabaseIO* expandedDB{nullptr};
+
+    std::tie(expanded, expandedDB) = expand_topology_files(generator, util, ioDB, dbType,
+                                                           db->get_property_manager(), step);
+    if(!expanded.empty()) {
+      std::cout << "Found file: " << expanded << std::endl;
+      locate_state(expandedDB, targetTime, loc);
+      delete expandedDB;
+    }
+    else {
+      found = false;
+    }
+  }
+}
+
+void locate_linear_multi_db_state(Ioss::DatabaseIO* db, double targetTime,
+                                  const std::string& ioDB, const std::string& dbType,
+                                  DatabaseState& loc)
+{
+  auto util = db->util();
+  FileNameGenerator generator = [](const std::string& baseFileName, unsigned step) {
```

Comment:
Maybe I'm misreading something here, but isn't this casting the type of a lambda into a std::function (or whatever FileNameGenerator is)?

Reply:
I believe this is standard where you construct a std::function from a lambda. It's not a cast